.github/workflows/publish-prisma-fmt-wasm.yml

```diff
@@ -0,0 +1,78 @@
+name: Build and publish @prisma/prisma-fmt-wasm
+
+on: workflow_dispatch # triggered manually for now
+
+jobs:
+  build:
+    name: Build and publish @prisma/prisma-fmt-wasm
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2.3.4
+
+    # Install nix
+    # Copy-pasted from https://github.com/marketplace/actions/install-nix
+    - uses: cachix/install-nix-action@v14
+      with:
+        install_url: https://nixos-nix-install-tests.cachix.org/serve/vij683ly7sl95nnhb67bdjjfabclr85m/install
+        install_options: '--tarball-url-prefix https://nixos-nix-install-tests.cachix.org/serve'
+        extra_nix_config: |
+          experimental-features = nix-command flakes
+          access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+
+    - name: Print input
+      env: { THE_INPUT: '${{ toJson(github.event.client_payload) }}' }
+      run: |
+        echo $THE_INPUT
+
+    # Update the cargo hash
+    - run: |
+        cd ./build-prisma-fmt
+        # Update dependencies (including engines).
+        nix run .#cargo update
+        # Set up build to fail and output the new hash.
+        sed -i 's/cargoSha256.*/cargoSha256 = fakeSha256;/' flake.nix
+        # Extract the new dependencies source hash
+        CARGO_HASH=`nix build 2>&1 > /dev/null | grep 'got:.*sha256' | sed 's/.*got:.*sha256/sha256/'`
+        echo "New cargo hash: $CARGO_HASH"
+        # Put the new cargo hash in flake.nix
+        sed -i "s/cargoSha256.*$/cargoSha256 = \"$CARGO_HASH\";/" flake.nix
+
+    # Build
+    - run: nix build ./build-prisma-fmt
+
+    # Update the npm package version
+    - run: |
+        cd ./build-prisma-fmt
+        nix run .#npm version patch
+
+    # Publish
+    - uses: actions/setup-node@v2
+      with:
+        node-version: '14.x'
+
+    - run: bash .github/workflows/setup.sh
+      env:
+        CI: true
+        NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+    - uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        # Optional but recommended, defaults to "Apply automatic changes"
+        commit_message: engines commit ${{ github.event.client_payload.commit }}
+
+        # Optional commit user and author settings
+        commit_user_name: prisma-bot
+        commit_user_email: prismabots@gmail.com
+        commit_author: prisma-bot <prismabots@gmail.com>
+
```

Comment:
```suggestion

```
Let's remove the auto commit it's going to fail often because of concurrency of commits

Reply:
We can't do without, because we are updating lockfiles and the npm package version.

Reply:
What kind of updates? (only npm package version?)
If we don't persist the npm package version and follow the current version scheme like done for the other packages we won't need to persist files actually

Reply:
Cool, that's one problem fewer. We have the `Cargo.lock` file and the `cargoSha256` checksum in the flake — if we can't persist these, the builds won't be reproducible and we won't have a trace of what we published.

Reply:
There is a build concurrency setting on GH actions, I don't know if it's possible to rely on that? It isn't in the GH workflow yet, but I imagined we would trigger this build after the engines wrapper build, so there would be no concurrency along that line.

Reply:
So you suggest that the Cargo.lock and cargoSha256 will change for every build?

Reply:
Yes, because there's a new engines version at every build.

Reply:
The engine version hash will be in the version of the published package so it's also ok to not persist the `Cargo.lock` here?

Reply:
It's different from the engines build, because the engines hash points to the git commit in engines that contains the `Cargo.lock`, whereas this build doesn't use the engines `Cargo.lock` but the one in this repo. We'd have to build in prisma-engines, which we tried and turned out to be complicated because the wasm package doesn't interact well with the regular build pipeline.

Reply:
For the record, my preferred alternative would be having the whole schema parser in a separate repo, with its versioning scheme and its own CI, and release the wasm package from there. Of course that's a much bigger change.

Reply:
Then we could actually have this in a separate repo with this GitHub workflow taking as input in a worflow_dispatch the version to publish and it could take that and publish and commit without any problem?

Reply:
I'd be happy with that. @janpio what do you think?

Reply:
Generally we want less repos, not more - so let's see if we can represent this in here somehow, possibly without the commit or some other way (branches, floating commits, whatever).

Reply:
After more discussion with @janpio , we're going to do that in a separate repo. I'll close this PR as soon as it's set up there, with a link.